src/streaming/text/DVBFonts.js

```diff
@@ -0,0 +1,258 @@
+            logger.debug(`Removal of fontFamily: ${font.fontFamily} was ${deleted ? 'successful' : 'unsuccessful'}`);
+        }
+    };
+
+    /**
+     * Check the attributes of a supplemental or essential property descriptor to establish if 
+     * it has the mandatory values for a dvb font download
+     * @param {object} attrs - property descriptor attributes
+     * @returns {boolean} true if mandatory attributes present
+     * @private
+     */
+    function _hasMandatoryDvbFontAttributes(attrs) {
+        if (
+            (attrs.value && attrs.value === '1') &&
+            (attrs.dvb_url && attrs.dvb_url.length > 0) && 
+            (attrs.dvb_fontFamily && attrs.dvb_fontFamily.length > 0) &&
+            (attrs.dvb_mimeType && (attrs.dvb_mimeType === Constants.OFF_MIMETYPE || attrs.dvb_mimeType === Constants.WOFF_MIMETYPE))
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Resolves a given font download URL.
+     * @param {string} fontUrl - URL as in the 'dvb:url' property
+     * @param {string} baseUrl - BaseURL for Adaptation Set 
+     * @returns {string} resolved URL
+     * @private
+     */
+    function _resolveFontUrl(fontUrl, baseUrl) {
+        if (urlUtils.isPathAbsolute(fontUrl)) {
+            return fontUrl;
+        } else if (urlUtils.isRelative(fontUrl)) {
+            if (baseUrl) {
+                return urlUtils.resolve(fontUrl, baseUrl);
+            } else {
+                return urlUtils.resolve(fontUrl);
+            }
+        } else {
+            return fontUrl; 
+        }
+    };
+
+    /**
+     * Updates the status of a given dvb font relative to whether it is loaded in the browser
+     * or if the download has failed
+     * @param {number} index - Index of font in dvbFontList
+     * @param {string} newStatus - Status value to update. Property of FONT_DOWNLOAD_STATUS
+     * @private
+     */
+    function _updateFontStatus(index, newStatus) {
+        const font = dvbFontList[index];
+        dvbFontList[index] = {...font, status: newStatus};
+    };
+
+    /**
+     * Adds all fonts to the dvb font list from all tracks
+     * @param {array} tracks - All text tracks
+     * @param {string} streamId - Id of the stream
+     */
+    function addFontsFromTracks(tracks, streamId) {
+        if (tracks && Array.isArray(tracks) && streamId) {
+            for (let i = 0; i < tracks.length; i++) {
+                let track = tracks[i];
+                _addFontFromTrack(track, streamId);
+            };
+        }
+    }
+
+    /**
+     * Initiate the download of a dvb custom font.
+     * The browser will neatly handle duplicate fonts
+     */
+    function downloadFonts() {
+        for (let i = 0; i < dvbFontList.length; i++) {
+            let font = dvbFontList[i];
+
+            document.fonts.add(font.fontFace);
+            eventBus.trigger(Events.DVB_FONT_DOWNLOAD_ADDED, font);
+
+            font.fontFace.load();
```

Comment:
Can we use the promise returned by `load()` directly instead of the one returned by `loaded`?